File: backend/src/main/java/com/sushishop/domain/Promotion.java
package com.sushishop.domain;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "promotions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = true)
public class Promotion extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String title;

    @Column(length = 250)
    private String description;

    @Column(nullable = false)
    private BigDecimal discountPercent;

    @Column(nullable = false)
    private LocalDateTime startDate;

    @Column(nullable = false)
    private LocalDateTime endDate;

    @ManyToMany
    @JoinTable(
            name = "promotion_products",
            joinColumns = @JoinColumn(name = "promotion_id"),
            inverseJoinColumns = @JoinColumn(name = "product_id")
    )
    private List<Product> products;

    @Builder.Default
    private boolean active = true;
}

File: frontend/src/types/promotion.ts
import type { ProductListResponse } from "./product"

export interface CreatePromotionRequest {
  title: string
  description?: string
  discountPercent: number
  startDate: string
  endDate: string
  productIds: number[]
}

export interface UpdatePromotionRequest {
  title?: string
  description?: string
  discountPercent?: number
  startDate?: string
  endDate?: string
  productIds?: number[]
}

export interface PromotionResponse {
  id: number
  slug: string
  title: string
  description?: string
  discountPercent: number
  startDate: string
  endDate: string
  active: boolean
  products: ProductListResponse[]
}